core/src/main/scala/com/softwaremill/react/kafka2/Consumer.scala

```diff
@@ -75,7 +89,7 @@ object ManualCommitConsumer {
   }
 }
 
-class ManualCommitConsumer[K, V](consumerProvider: ConsumerProvider[K, V])
+class ManualCommitConsumer[K, V](consumerProvider: () => KafkaConsumer[K, V])
```

Comment:
thanks for clarifying
`offset` and manual partitions could be part of the settings
mocking of the `KafkaKonsumer` is an interesting point, but I think that should be done by replacing the `Source` itself, instead of mocking the rather involved `KafkaConsumer` api